components/build/src/lib.rs

```diff
@@ -147,7 +149,16 @@ fn generate_mod_rs<P: AsRef<Path>, Q: AsRef<Path>>(
             }
         })
     };
-    let proto_files = include_proto_files(proto_files);
+
+    // To avoid cases where input sources are also added as includes, use only
+    // unique paths.
+    let proto_files: Vec<PathBuf> = [proto_files, includes]
+        .concat()
+        .into_iter()
+        .unique()
```

Comment:
I think you can change signature of `include_proto_files` to `IntoIter<..>` and use a `HashSet` here (so we can get rid of `itertools` dependency). `unique()` [uses a set with clones of added items](https://docs.rs/itertools/0.8.0/itertools/trait.Itertools.html#method.unique) inside anyway.

Reply:
Agree with @popzxc. One more extra dependency for a one method. It's not worth it. Let's do it with `HashSet`.